mountpoint-s3-fs/src/config.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::Context as _;
+use futures::executor::block_on;
+use mountpoint_s3_client::ObjectClient;
+
+use crate::data_cache::{DataCacheConfig, DiskDataCache, ExpressDataCache, MultilevelDataCache};
+use crate::fuse::config::FuseSessionConfig;
+use crate::fuse::session::FuseSession;
+use crate::fuse::S3FuseFilesystem;
+use crate::prefetch::{Prefetcher, PrefetcherBuilder};
+use crate::s3::config::S3Path;
+use crate::sync::Arc;
+use crate::{Runtime, S3Filesystem, S3FilesystemConfig};
+
+/// Configuration for a Mountpoint session
+#[derive(Debug)]
+pub struct MountpointConfig {
+    fuse_session_config: FuseSessionConfig,
+    data_cache_config: DataCacheConfig,
+    filesystem_config: S3FilesystemConfig,
+}
+
+impl MountpointConfig {
+    pub fn new(fuse_session_config: FuseSessionConfig) -> Self {
```

Comment:
Why is `fuse_session_config` the only argument that's required here?

Reply:
Updated